ktoml-core/src/commonMain/kotlin/com/akuleshov7/ktoml/decoders/TomlMainDecoder.kt

```diff
@@ -173,51 +174,63 @@ public class TomlMainDecoder(
         }
     }
 
+    /**
+     * A hack that comes from a compiler plugin to process Inline (value) classes
+     */
+    override fun decodeInline(inlineDescriptor: SerialDescriptor): Decoder =
+            iterateOverStructure(inlineDescriptor, true)
+
     /**
      * this method does all the iteration logic for processing code structures and collections
      * treat it as an !entry point! and the orchestrator of the decoding
      */
```

Comment:
very strange that we see changes here... it looks like they are coming from https://github.com/akuleshov7/ktoml/pull/133

Reply:
I rebased when new changes were made to the main branch, which seems to be the wrong thing to do. Should I have merged the main branch in this one?

Reply:
Right now everything looks fine actually. I simply merged changes from ktoml main branch to your branch.